app/javascript/mastodon/features/audio/index.js

```diff
@@ -280,7 +281,7 @@ class Audio extends React.PureComponent {
 
     if(!isNaN(x)) {
       this.setState({ volume: x }, () => {
-        this.audio.volume = x;
+        this.gainNode.gain.value = this.state.muted ? 0 : x;
```

Comment:
Sorry I missed that, but it needs to check that `this.gainNode` is defined, which might not be the case if the user attempts to change the volume before starting playback.

```suggestion
        if (this.gainNode) {
          this.gainNode.gain.value = this.state.muted ? 0 : x;
        }
```